server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -310,6 +297,8 @@ static boolean executeInternal(SearchContext searchContext) throws QueryPhaseExe
                 queryResult.nodeQueueSize(rExecutor.getCurrentQueueSize());
                 queryResult.serviceTimeEWMA((long) rExecutor.getTaskExecutionEWMA());
             }
+            // Search phase has finished, no longer need to check for timeout
+            searcher.unsetCheckTimeout();
```

Comment:
Why do we need to do this?

Reply:
So that we don't experience a timeout during the aggregation phase. (there were integ tests that were failing because of this). I changed the comment to mention it.

Reply:
We should do this on the cancellable object directly. The searcher shouldn't be able to unset this way. See my comment on the context searcher.